### PR TITLE
Build container image on Python Matter server release

### PR DESCRIPTION
**Pull request by agners**
This adds a `Dockerfile` along with GitHub Action rules to build an official container image for the Python Matter server.

Fixes: #308

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,94 @@ This project implements a Matter Controller Server over WebSockets using the [of
...
 The goal of this project is primary to have Matter support in Home Assistant but its universal approach makes it suitable to be used in other projects too.
 
-This repository is for development only (so not for enduser support). For enabling Matter support within Home Assistant, please refer to the [Home Assistant documentation](https://www.home-assistant.io/integrations/matter/).
+We strongly recommend to use Home Assistant OS along with the official Matter
+Server add-on to use Matter with Home Assistant. The Matter integration
+automatically installs the Python Matter Server add-on. Please refer to the
+[Home Assistant documentation](https://www.home-assistant.io/integrations/matter/).
+Home Assistant OS has been tested and tuned to be used with Matter and Thead,
+which makes this combination the best tested and largely worry free
+environement.
```
Comment (marcelveldt) [inline README.md:13]:
```suggestion
environment.
```

File: README.md [hunk 2]
```diff
@@ -4,7 +4,94 @@ This project implements a Matter Controller Server over WebSockets using the [of
 
 The goal of this project is primary to have Matter support in Home Assistant but its universal approach makes it suitable to be used in other projects too.
 
-This repository is for development only (so not for enduser support). For enabling Matter support within Home Assistant, please refer to the [Home Assistant documentation](https://www.home-assistant.io/integrations/matter/).
```
Comment (marcelveldt) [inline README.md:7]: we should somehow keep this text in as this still applies, this is the repo for the matter server library and for HA enduser support, people should go to HA.

File: README.md [hunk 3]
```diff
@@ -4,7 +4,94 @@ This project implements a Matter Controller Server over WebSockets using the [of
...
+
+For communication through Thread border routers which are not running on the same
+host as the Matter Controller server to work, IPv6 routing needs to be properly
+working. IPv6 routing is largely setup automatically through the IPv6 Neighbor
+Discovery Protocol, specifically the Route Information Options (RIO). However,
+if IPv6 ND RIO's are processed, and processed correctly depends on the network
+management software your system is using, there may be bugs and cavats in
+processing this Route Information Options.
+
+In general, make sure the kernel option `CONFIG_IPV6_ROUTER_PREF` is enabeld and
```
Comment (marcelveldt) [inline README.md:40]:
```suggestion
In general, make sure the kernel option `CONFIG_IPV6_ROUTER_PREF` is enabled and
```

File: README.md [hunk 4]
```diff
@@ -4,7 +4,94 @@ This project implements a Matter Controller Server over WebSockets using the [of
...
+that IPv6 forwarding is disabled (sysctl variable `net.ipv6.conf.all.forwarding`).
+If IPv6 forwarding is enabled, the Linux kernel doesn't employ reachability
+probing (RFC 4191), which can lead to longer outages (up to 30min) until
+network changes are detected.
+
+If you are using NetworkManager, make sure to use at least NetworkManager 1.42.
+Previous versions loose track of routes and stale routs can lead to unreachable
```
Comment (marcelveldt) [inline README.md:47]:
```suggestion
Previous versions loose track of routes and stale routes can lead to unreachable
```

File: README.md [hunk 5]
```diff
@@ -4,7 +4,94 @@ This project implements a Matter Controller Server over WebSockets using the [of
...
+Thread devices. All current released NetworkManager versions can't handle
+multiple routes to the same network properly. This means if you have multiple
+Thread border routers, the fallback won't work immeaditly (see [NetworkManager
```
Comment (marcelveldt) [inline README.md:50]:
```suggestion
Thread border routers, the fallback won't work immediately (see [NetworkManager
```

File: README.md [hunk 6]
```diff
@@ -4,7 +4,94 @@ This project implements a Matter Controller Server over WebSockets using the [of
...
+```
+mkdir data
+docker run -d \
+  --name matter-server \
+  --restart=unless-stopped \
+  --security-opt apparmor=unconfined \
+  -v $(pwd)/data:/data \
+  -v /run/dbus:/run/dbus:ro \
+  --network=host \
+  ghcr.io/agners/python-matter-server:stable
```
Comment (marcelveldt) [inline README.md:86]: shouldn't this be home-assistant-libs ?

File: compose.yml [hunk 7]
```diff
@@ -2,9 +2,6 @@ version: "3.8"
 services:
   # python-matter-server
   matter-server:
-    build:
-      context: ./
-      dockerfile: Dockerfile.dev
```
Comment (marcelveldt) [inline compose.yml:7]: do we also need to delete the dockerfile.dev ?

File: README.md (hunk 4 above)
Comment (digiblur) [inline README.md:47]: lose track not loose

File: compose.yml (hunk 7 above)
Comment (agners) [reply compose.yml:7]: I've converted the `compose.yml` to use the new container.

However, for development the container *could* still be useful. On the other hand, probably a dev container would be more appropriate then. Thoughts? I am fine removing it.

File: README.md (hunk 6 above)
Comment (agners) [reply README.md:86]:
```suggestion
  ghcr.io/home-assistant-libs/python-matter-server:stable
```

File: compose.yml (hunk 7 above)
Comment (MartinHjelmare) [reply compose.yml:7]: I think we should keep it until we have an alternative in devcontainer.

File: README.md [hunk 8]
```diff
@@ -4,7 +4,94 @@ This project implements a Matter Controller Server over WebSockets using the [of
...
+We strongly recommend to use Home Assistant OS along with the official Matter
+Server add-on to use Matter with Home Assistant. The Matter integration
+automatically installs the Python Matter Server add-on. Please refer to the
+[Home Assistant documentation](https://www.home-assistant.io/integrations/matter/).
+Home Assistant OS has been tested and tuned to be used with Matter and Thead,
+which makes this combination the best tested and largely worry free
+environment.
+
+If you still prefer a self-managed container installation, we do offer an
+official container image. Please keep in mind that you might expierence
```
Comment (MartinHjelmare) [inline README.md:16]:
```suggestion
official container image. Please keep in mind that you might experience
```

File: README.md [hunk 9]
```diff
@@ -4,7 +4,94 @@ This project implements a Matter Controller Server over WebSockets using the [of
...
+The host network interface needs IPv6 support enabled.
+
+## Requirements to communicate with Thread devices through Thread border routers
+
+For communication through Thread border routers which are not running on the same
+host as the Matter Controller server to work, IPv6 routing needs to be properly
+working. IPv6 routing is largely setup automatically through the IPv6 Neighbor
+Discovery Protocol, specifically the Route Information Options (RIO). However,
+if IPv6 ND RIO's are processed, and processed correctly depends on the network
+management software your system is using, there may be bugs and cavats in
```
Comment (MartinHjelmare) [inline README.md:37]:
```suggestion
management software your system is using. There may be bugs and cavats in
```

File: README.md [hunk 10]
```diff
@@ -4,7 +4,94 @@ This project implements a Matter Controller Server over WebSockets using the [of
...
+issue #1232](https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/1232)).
+
+We currently don't have experience with systemd-networkd. It seems to have its
+own IPv6 Neighbor Discovery Protocol handling.
+
+If you don't use NetworkManager or systemd-networkd, you can use the kernel's
+IPv6 Neighbor Discovery Protocol handling.
+
+Make sure the kernel options `CONFIG_IPV6_ROUTE_INFO` is enabled and the
+following sysctl variables:
```
Comment (MartinHjelmare) [inline README.md:60]:
```suggestion
following sysctl variables are set:
```

File: README.md [hunk 11]
```diff
@@ -4,7 +4,94 @@ This project implements a Matter Controller Server over WebSockets using the [of
...
+
+```
+sysctl -w net.ipv6.conf.wlan0.accept_ra=1
+sysctl -w net.ipv6.conf.wlan0.accept_ra_rt_info_max_plen=64
+```
+
+If your system has IPv6 forwarding enabled (not recommended, see above), you'll
+have to use 2 for the accept_ra variable. See also the [Thread Border Router - Bidirectional IPv6 Connectivity and DNS-Based Service Discovery codelab](https://openthread.io/codelabs/openthread-border-router#6).
```
Comment (MartinHjelmare) [inline README.md:68]:
```suggestion
have to use `2` for the accept_ra variable. See also the [Thread Border Router - Bidirectional IPv6 Connectivity and DNS-Based Service Discovery codelab](https://openthread.io/codelabs/openthread-border-router#6).
```

File: README.md (hunk 2 above)
Comment (agners) [reply README.md:7]: The reason I've removed it is that the same readme is shown in the package repository. So this could be interpreted somewhat wrong.

**Review by MartinHjelmare — APPROVED** [review]

**Review by marcelveldt — APPROVED** [review]

**Pull request merged**